RxSwift/Observables/Buffer.swift

```diff
@@ -136,3 +136,131 @@ final private class BufferTimeCountSink<Element, Observer: ObserverType>
         nextTimer.setDisposable(disposable)
     }
 }
+
+extension ObservableType {
+    
+    /**
+     Projects each element of an observable sequence info a buffer that's sent out when the trigger Observable emits a .next event or a .complete event, if the accumulated buffer's count
+     is over zero.
+     
+     - seealso: [buffer operator on reactivex.io](http://reactivex.io/documentation/operators/buffer.html)
+     
+     - parameter trigger: Observable that will act as a boundary trigger between each window.
+     - returns: An observable sequence of buffers.
+     */
+    public func buffer<TriggerElement>(trigger: Observable<TriggerElement>) -> Observable<[Element]> {
+        return BufferTrigger(source: self.asObservable(), trigger: trigger)
+    }
+    
+    /**
+     Projects each element of an observable sequence info a buffer that's sent out when the trigger Observable emits a .next event or a .complete event, if the accumulated buffer's count
+     is over zero.
+     
+     A useful real-world example of this is when you have a resource that needs to be updated depending on some asynchronous tasks and you want to accumulate all of those tasks's results
+     and just PUT/PATCH the remote resource once per batch of results instead of once per result.
+     
+     - seealso: [buffer operator on reactivex.io](http://reactivex.io/documentation/operators/buffer.html)
+     
+     - parameter debounce: Amount of time to debounce the source as the boundary trigger between each window.
+     - parameter scheduler: Scheduler to run debouncing on.
+     - returns: An observable sequence of buffers.
+     */
+    public func buffer(debounce: RxTimeInterval, scheduler: SchedulerType) -> Observable<[Element]> {
+        let shared = self.share()
+        return shared.buffer(trigger: shared.debounce(debounce, scheduler: scheduler))
+    }
+}
+
+final fileprivate class BufferTrigger<Element, TriggerElement> : Producer<[Element]> {
+    
+    fileprivate let _source: Observable<Element>
+    fileprivate let _trigger: Observable<TriggerElement>
+    
+    init(source: Observable<Element>, trigger: Observable<TriggerElement>) {
+        _source = source
+        _trigger = trigger
+    }
+    
+    override func run<O : ObserverType>(_ observer: O, cancel: Cancelable) -> (sink: Disposable, subscription: Disposable) where O.Element == [Element] {
```

Comment:
```suggestion
    override func run<O : ObserverType>(_ observer: O, cancel: Cancelable)
     -> (sink: Disposable, subscription: Disposable) where O.Element == [Element] {
```

A bit of breathing room :)